### DISCUSSION TITLE
webpack.config.js in another directory than Cypress - how to change where it looks for support files?

### DISCUSSION THREAD

**Discussion by pojocariba** [cypress-io/cypress#25566] [Questions and Help]
I get Module not found: Error: Can't resolve 'sdk/theme/themes/[projectName]/src/js/cypress/support/component-index.html' because i have cypress located at sdk/tests/[projectName]/. How can i fix this problem without changing my project structure? 

for better understanding, this is my project structure:
![Screenshot from 2023-01-25 11-15-26](https://user-images.githubusercontent.com/61711192/214538245-a35cdfe0-45f8-4fe6-ad90-2133fac2532c.jpg)

and content of my webpack.config file is:
```js
const path = require('path');
const srcPath = path.resolve(__dirname + '/src/js')

module.exports = {
    context: srcPath,
    mode: 'production',
    watch: false,
    entry: {
        main: srcPath + '/styleguide/index.js'
    },
    output: {
        filename: 'styleguide-webpack.js'
    },
    devtool: 'cheap-module-eval-source-map',
    module: {
        rules: [
            {
                test: /\.js$/,
                exclude: /node_modules/,
                use: {
                    loader: 'babel-loader',
                    options: {
                        presets: ['@babel/preset-env']
                    }
                }
            }
        ]
    },
    plugins: []
```

I have also tried to put another webpack.config.js file in cypress directory, to fix a problem but the problem persists. Here is content of that file:
```js
module.exports = {
    entry: {
        cypress : "./",
      babel:  "./../../theme/themes/a1s-ecom-theme/webpack.config.babel"
    },

    output: {
        path: './dist',
        filename: 'bundle.js'
    }
};
```
Hope anyone can help c:

**Comment by portyu9** [comment]
Hi @pojocariba 

You should not create a second Webpack entry for Cypress.

There are two separate paths involved here:

- Cypress's own Component Testing files (`component-index.html`, support file)
- your application's Webpack configuration

Your application Webpack config sets:

```js
context: srcPath
```

where `srcPath` is inside `sdk/theme/.../src/js`. That is a strong clue as to why a relative Cypress path is ending up below that directory.

Configure the Cypress index/support paths explicitly, and pass the external Webpack config as the Component Testing bundler configuration.

Conceptually:

```js
const { defineConfig } = require('cypress')
const path = require('path')

const webpackConfig = require(
  '../../theme/themes/[projectName]/webpack.config.babel'
)

module.exports = defineConfig({
  component: {
    indexHtmlFile: path.resolve(
      __dirname,
      'cypress/support/component-index.html'
    ),

    supportFile: path.resolve(
      __dirname,
      'cypress/support/component.ts'
    ),

    devServer: {
      // keep the framework you already use here
      framework: 'react',
      bundler: 'webpack',
      webpackConfig,
    },
  },
})
```

Replace `react` with your actual configured Component Testing framework if this is not a React project.

The important part is that `indexHtmlFile` points to the Cypress project's HTML file, while `webpackConfig` points to the application's Webpack configuration in the other directory.

This attempt:

```js
entry: {
  cypress: './',
  babel: './../../theme/.../webpack.config.babel'
}
```

does not configure Cypress's support/index locations. A Webpack config file is configuration, not an application entry module, so adding it as an `entry` is the wrong layer to solve this.

Current Cypress can also automatically find Webpack configs by searching upward from the project root, but because yours lives in a separate sibling tree, passing `webpackConfig` explicitly is the clearest solution.

Please mark this answer as accepted if it helped, thank you.
